enhancements/20220525-system-logging.md

```diff
@@ -0,0 +1,98 @@
+# System Logging
+
+## Summary
+
+We need to be able to aggregate all cluster pod logs outside the cluster, to allow for collecting and analyzing those logs to aid in debugging and harvester management.
+
+### Related Issues
+
+- https://github.com/harvester/harvester/issues/577
+
+## Motivation
+
+### Goals
+
+List the specific goals of the enhancement. How will we know that this has succeeded?
+
+- The user can aggregate harvester logs in a centralized location
+- The user can view the aggregated harvester logs outside the cluster
+
+### Non-goals
+
+- this enhancement does not cover integration into the harvester / rancher UI (but ideally this will eventually be implemented as well)
+
+## Proposal
+
+We will deploy a new `ManagedChart` in the [Harvester Installer](https://github.com/harvester/harvester-installer) to
+install. The `ManagedChart` will deploy a [`ClusterFlow`](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/flow/) to select and 
+aggregate interesting logs, which can be managed by a new [`ClusterOutput`](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/output/) 
+crd. The `ClusterOutput` can then be configured by settings from the harvester UI.
```

Comment:
From the [issue](https://github.com/harvester/harvester/issues/577):
> Log in the OS. E.g. major system logs under /var/log/

Can the framework handle this too?

Reply:
I will need to test, but based on documentation, we should be able to mount host paths into the fluentbit pods which are scheduled as a DS. 

https://banzaicloud.com/docs/one-eye/logging-operator/configuration/crds/v1beta1/fluentbit_types/#volumemount

example: https://banzaicloud.com/docs/one-eye/logging-operator/configuration/fluentbit/#volume-mount

Reply:
`rancher-logging` values.yaml has such field, need to check if `/var/log/` can also be collected.
```
# If your additional logging sources collect logs from systemd configure the systemd log path here
systemdLogPath: "/run/log/journal"
```